### PR TITLE
[TypeScript SDK] Add support for txn deserialization

### PR DESCRIPTION
**Pull request by 666lcz**

### CODE REVIEW

File: sdk/typescript/src/signers/txn-data-serializers/local-txn-data-serializer.ts [hunk 1]
```diff
@@ -352,6 +356,123 @@ export class LocalTxnDataSerializer implements TxnDataSerializer {
...
+      };
+    } else if ('Call' in tx) {
+      return {
+        kind: 'moveCall',
+        data: {
+          packageObjectId: tx.Call.package.objectId,
+          module: tx.Call.module,
+          function: tx.Call.function,
+          typeArguments: tx.Call.typeArguments,
+          arguments: await new CallArgSerializer(
```
Comment (Jordan-Mysten) [inline sdk/typescript/src/signers/txn-data-serializers/local-txn-data-serializer.ts:436]: I wonder if this should be an option in the event that you just want to partially deserialize calls.

**Review by gegaowp — APPROVED** [review]
good move!

File: sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts [hunk 2]
```diff
@@ -74,6 +66,41 @@ export class CallArgSerializer {
...
+    );
+
+    return Promise.all(
+      userParams.map(async (param, i) =>
+        this.deserializeCallArg(param, txn.Call.arguments[i])
+      )
+    );
+  }
+
+  private async getUserParams(
```
Comment (Jordan-Mysten) [inline sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts:86]: I don't love this function name but not a blocker at all.

**Review by Jordan-Mysten — APPROVED** [review]
Looks good to me. Really hope to eventually build out a `Transaction` class to encapsulate this in the near future.

File: sdk/typescript/src/signers/txn-data-serializers/local-txn-data-serializer.ts (hunk 1 above)
Comment (666lcz) [reply sdk/typescript/src/signers/txn-data-serializers/local-txn-data-serializer.ts:436]: In that case they can use `deserializeTransactionBytesToTransactionData` on line 376

**Pull request merged**